content/issue-1/vecdeque-resize-optimization.md

```diff
@@ -0,0 +1,219 @@
+> This is the first article on **pr-demystifying** topics. Each article labeled **pr-demystifying** will attempt to demystify the details behind the PR.
+
+While browsing the rust-lang repository's list of pull requests last month, I came across PR [#104435], titled `VecDeque::resize should re-use the buffer in the passed-in element`. This PR caught my attention because it seemed interesting and I wanted to understand more about it. I began to wonder why it was necessary to optimize `VecDeque::resize()` and how the old version might be lacking. I also wanted to know how the author had optimized the new version. After delving into the code in the PR, I was able to gain a deeper understanding of these issues.
+
+## VecDeque::resize()
+
+Firstly, let's get familiar with the [VecDeque::resize()](https://doc.rust-lang.org/std/collections/struct.VecDeque.html#method.resize).
+
+> Modifies the deque in-place so that len() is equal to new_len, either by removing excess elements from the back or by appending clones of value to the back.
+
+```rs
+use std::collections::VecDeque;
+
+let mut buf = VecDeque::new();
+buf.push_back(5);
+buf.push_back(10);
+buf.push_back(15);
+assert_eq!(buf, [5, 10, 15]);
+
+buf.resize(2, 0);
+assert_eq!(buf, [5, 10]);
+
+buf.resize(5, 20);
+assert_eq!(buf, [5, 10, 20, 20, 20]);
+```
+
+The `VecDeque::resize()` API is simple to use. It takes two arguments: the new length to which the `VecDeque` should be resized, and a default value to use for any new elements that are added to the `VecDeque` when it expands.
```

Comment:
I wouldn't say "default" here because that could be confused with `.resize_with(Default::default)`.